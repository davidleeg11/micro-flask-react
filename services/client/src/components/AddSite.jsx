import React from 'react';

const AddSite = (props) => {
  return (
    <form onSubmit={(event) => props.addSite(event)}>
      <div className="field">
        <input
          name="site"
          className="input is-large"
          type="text"
          placeholder="Enter a site"
          required
          value={props.site}  // new
          onChange={props.handleChange}  // new
        />
      </div>
      <input
        type="submit"
        className="button is-primary is-large is-fullwidth"
        value="Submit"
      />
    </form>
  )
};

export default AddSite;
