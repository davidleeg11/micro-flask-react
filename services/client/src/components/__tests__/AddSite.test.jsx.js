import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddSite from '../AddSite';

test('AddSite renders properly', () => {
  const wrapper = shallow(<AddSite/>);
  const element = wrapper.find('form');
  expect(element.find('input').length).toBe(2);
  expect(element.find('input').get(0).props.name).toBe('site');
  expect(element.find('input').get(1).props.type).toBe('submit');
});

test('AddSite renders a snapshot properly', () => {
  const tree = renderer.create(<AddSite/>).toJSON();
  expect(tree).toMatchSnapshot();
});
