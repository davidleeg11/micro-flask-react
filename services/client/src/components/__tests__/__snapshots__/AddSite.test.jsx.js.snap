// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSite renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="site"
      onChange={undefined}
      placeholder="Enter a site"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <input
    className="button is-primary is-large is-fullwidth"
    type="submit"
    value="Submit"
  />
</form>
`;
